pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -345,4 +404,26 @@ func DeleteURLsConfigMap(revised *corev1.ConfigMap, crName string, newURLs []cor
 			}
 		}
 	}
+
+	return deletedPeers
+}
+
+//RemoveSearchPeers removes deleted search peers and updates etc/system/local/distsearch.conf
+func RemoveSearchPeers(c splcommon.ControllerClient, cr splcommon.MetaObject, spec enterprisev1.CommonSplunkSpec, deletedPeers []string) error {
+	cmPodName := fmt.Sprintf("splunk-%s-monitoring-console-0", cr.GetNamespace())
+	adminPwd, err := splutil.GetSpecificSecretTokenFromPod(c, cmPodName, cr.GetNamespace(), "password")
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(deletedPeers); i++ {
+		command := fmt.Sprintf("/opt/splunk/bin/splunk remove search-server %s:8089 -auth admin:%s", deletedPeers[i], adminPwd)
+		_, _, err := splutil.PodExecCommand(c, cmPodName, cr.GetNamespace(), []string{"/bin/sh"}, command, false, false)
+		if err != nil {
+			if !spec.Mock {
+				return err
+			}
+		}
```

Comment:
When we are in this loop, and I think failing for any of the peers in the list will make us to restart doing the podExec commands again from the beginning. So, I think it will be great if we can validate and make sure that, failing this command for a non-existing host should not put an error condition, pushing us to an infinite loop.

Reply:
Wouldn't the `deletedPeers` be updated in every reconcile loop? If yes, then we are fine but if not, then we definitely need to keep a track of already removed peers.

Reply:
Yes, deleted peers get updated in every reconcile loop.

Reply:
I think it's a good idea to keep a track because in case we try to delete a peer that is deleted we may stuck in error loop forever. In the case of deleting multiple search servers together, we might face this. I should take care of this

Reply:
I guess, we can do a quick test to see the behavior, and if required, then only we can take care.